org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/ReferencesHandler.java

```diff
@@ -149,6 +161,85 @@ public void acceptSearchMatch(SearchMatch match) throws CoreException {
 						if (cf != null && cf.getSourceRange() != null) {
 							location = JDTUtils.toLocation(cf, match.getOffset(), match.getLength());
 						}
+						if (cf != null && location == null) {
+							ContentProviderManager contentProvider = JavaLanguageServerPlugin.getContentProviderManager();
+							String contents = contentProvider.getSource(cf, new NullProgressMonitor());
```

Comment:
Another idea is to add a preference like `java.references.includeDecompiledSources` to control whether to include the decompiled sources when finding references. The default value can be true. This is similar as `java.references.includeAccessors`.

Reply:
Fixed. See https://github.com/redhat-developer/vscode-java/pull/1773